tools/upip.py

```diff
@@ -195,6 +198,7 @@ def install_pkg(pkg_spec, install_path):
         f1.close()
     del f3
     del f2
+    del idx
```

Comment:
No need to del idx.  f2 and f3 are deleted only because they are files and need to be cleaned up.